readthedocs/projects/tasks.py

```diff
@@ -1417,6 +1417,28 @@ def warn(self, msg):
     delete_queryset.delete()
 
 
+@app.task(max_retries=3, default_retry_delay=7 * 60)
+def clean_build_task(version_id):
+    """Clean the files used in the build of the given version."""
+    version = Version.objects.get_object_or_log(pk=version_id)
+    if (
+        not version or
+        not version.project.has_feature(Feature.CLEAN_AFTER_BUILD)
+    ):
+        return
+    del_dirs = [
+        os.path.join(version.project.doc_path, dir_, version.slug)
+        for dir_ in ('checkouts', 'envs', 'conda')
+    ]
+    # TODO: the max_lock_age can be lower than the default
```

Comment:
Actually, default is fine. Because a build can take till x miinutes to timeout. If this task is triggered, it would start to delete everything and the build will fail, when maybe it only needed some more minutes.